core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -535,7 +536,7 @@ class GroupMetadataManager(val brokerId: Int,
 
     val numExpiredOffsetsRemoved = inWriteLock(offsetExpireLock) {
       val expiredOffsets = offsetsCache.filter { case (groupTopicPartition, offsetAndMetadata) =>
-        offsetAndMetadata.expireTimestamp < startMs
+        offsetAndMetadata.expireTimestamp < startMs || !metadataCache.contains(groupTopicPartition.topicPartition.topic)
```

Comment:
I don't think trying to avoid a read lock acquisition for a thread that by default runs every 10 minutes is really worth adding any new code.